kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,210 @@
+    Model& rModel
+    ) : mrModel(rModel)
+{
+    // Receive json string with materials properties, create Parameters
+    Parameters materials(rParametersName);
+
+    GetPropertyBlock(materials);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::GetPropertyBlock(Parameters Materials)
+{
+    KRATOS_INFO("Read materials") << "Started" << std::endl;
+    for (auto i = 0; i < Materials["properties"].size(); ++i) {
+        Parameters material = Materials["properties"].GetArrayItem(i);
+        AssignPropertyBlock(material);
+    }
+    KRATOS_INFO("Read materials") << "Finished" << std::endl;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::TrimComponentName(std::string& rLine){
+    std::stringstream ss(rLine);
+    std::size_t counter = 0;
+    while (std::getline(ss, rLine, '.')){counter++;}
+    if (counter > 1)
+        KRATOS_WARNING("Read materials") << "Ignoring module information for component " << rLine << std::endl;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
+{
+    // Get the properties for the specified model part.
+    ModelPart& r_model_part = mrModel.GetModelPart(Data["model_part_name"].GetString());
+    const IndexType property_id = Data["properties_id"].GetInt();
+    const IndexType mesh_id = 0;
+    Properties::Pointer p_prop = r_model_part.pGetProperties(property_id, mesh_id);
+
+    // Compute the size using the iterators
+    std::size_t variables_size = 0;
+    for(auto it=Data["Material"]["Variables"].begin(); it!=Data["Material"]["Variables"].end(); ++it)
+        variables_size++;
+    
+    std::size_t tables_size = 0;
+    for(auto it=Data["Material"]["Tables"].begin(); it!=Data["Material"]["Tables"].end(); ++it)
+        tables_size++;
+    
+    KRATOS_WARNING_IF("Read materials", variables_size > 0 && p_prop->HasVariables())
+        << "Property " << std::to_string(property_id) << " already has variables." << std::endl;
+    KRATOS_WARNING_IF("Read materials", tables_size > 0 && p_prop->HasTables())
+        << "Property " << std::to_string(property_id) << " already has tables." << std::endl;
+
+    // Assign the p_properties to the model part's elements and conditions.
+    auto& r_elements_array = r_model_part.Elements();
+    auto& r_conditions_array = r_model_part.Conditions();
+
+    #pragma omp parallel for
+    for(int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+        auto it_elem = r_elements_array.begin() + i;
+        it_elem->SetProperties(p_prop);
+    }
```

Comment:
Why you set properties by the model_part? This seems to be a important restriction respect to the current use of Properties per element and conditions.

I was expecting that by reading the material you would update the properties without affecting to whom they are assigned.

Reply:
This process is mimicking what we are currently doing with the read_materials_process.py. What do you suggest, taking into account that the elements and conditions may not have assigned any property at all in that moment (this is done just after reading the model part)